src/Tools/dotnet-monitor/Profiler/ProfilerService.cs

```diff
@@ -181,6 +182,20 @@ private async Task<string> CalculateRuntimeIdentifierAsync(DiagnosticsClient cli
             return runtimeIdentifier;
         }
 
+        private bool IsMutatingProfilerNeeded(IEndpointInfo endpointInfo)
```

Comment:
Consider inlining this with the caller and then log something when a feature callers for the mutating profiler but it is not supported thus will not be loaded.